core/drand_beacon.go

```diff
@@ -139,7 +139,9 @@ func (bp *BeaconProcess) Load() (bool, error) {
 		return false, fmt.Errorf("could not restore beacon info for the given identity - this can happen if you updated the group file manually")
 	}
 	bp.index = int(thisBeacon.Index)
+	bp.state.Lock()
```

Comment:
Shouldn't we also guard the index change?
```suggestion
	bp.state.Lock()
	bp.index = int(thisBeacon.Index)
```